src/pages/clubs/index.js

```diff
@@ -50,14 +50,19 @@ class Clubs extends React.PureComponent {
 
   // State management functions. Used by children components.
   handleToggleMap = () => {
-    this.setState({
+    this.setState(state => ({
       view: this.state.view === "map" ? "list" : "map",
-      number: this.state.view === "map" ? this.state.listClubNumber : 100,
+      number:
+        this.state.view === "map"
```

Comment:
use state here, not this.state (that's why I suggested the previous change). This way we can avoid stale data and keep state updates clean